packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/method_channel/method_channel_query.dart

```diff
@@ -19,19 +17,20 @@ import 'utils/exception.dart';
 /// communicate with Firebase plugins.
 class MethodChannelQuery extends QueryPlatform {
   /// Flags whether the current query is for a collection group.
+  @override
   final bool isCollectionGroupQuery;
 
   /// Create a [MethodChannelQuery] from a [path] and optional [parameters]
   MethodChannelQuery(
     FirebaseFirestorePlatform _firestore,
     String path, {
-    /*required*/ Map<String, dynamic> /*!*/ parameters,
+    Map<String, dynamic>? parameters,
```

Comment:
Why is this optional when the comment said it should be required?

Reply:
If you read the above comment, it says the `/// Create a [MethodChannelQuery] from a [path] and optional [parameters]`.

The superclass also initialises parameters [here](https://github.com/FirebaseExtended/flutterfire/blob/master/packages/cloud_firestore/cloud_firestore_platform_interface/lib/src/platform_interface/platform_interface_query.dart#L33).

Reply:
I think this is an erroneous nullsafety code hint. It should be optional based on code docs & super providing detailed details.